src/testers.jl

```diff
@@ -183,16 +183,16 @@ All keyword arguments except for `fdm` and `fkwargs` are passed to `isapprox`.
 function frule_test(f, xẋs::Tuple{Any, Any}...; rtol=1e-9, atol=1e-9, fdm=_fdm, fkwargs=NamedTuple(), kwargs...)
     _ensure_not_running_on_functor(f, "frule_test")
     xs, ẋs = first.(xẋs), last.(xẋs)
-    Ω_ad, dΩ_ad = frule((NO_FIELDS, ẋs...), f, xs...; fkwargs...)
-    Ω = f(xs...; fkwargs...)
+    Ω_ad, dΩ_ad = frule((NO_FIELDS, deepcopy(ẋs)...), f, deepcopy(xs)...; deepcopy(fkwargs)...)
+    Ω = f(deepcopy(xs)...; deepcopy(fkwargs)...)
```

Comment:
Should we also be `deepcopy`ing `f` here? If it has parameters, could they not conceivably be mutated during evaluation?

```suggestion
    Ω_ad, dΩ_ad = frule((NO_FIELDS, deepcopy(ẋs)...), deepcopy(f), deepcopy(xs)...; deepcopy(fkwargs)...)
    Ω = deepcopy(f)(deepcopy(xs)...; deepcopy(fkwargs)...)
```

Reply:
Good point, I'll update the test accordingly.